contracts/tests/tests/governor/test_execute.rs

```diff
@@ -91,6 +92,76 @@ fn test_execute() {
     );
 }
 
+#[test]
+fn test_execute_multi_level_sub_auth() {
+    let e = Env::default();
+    e.set_default_info();
+    e.mock_all_auths();
+    e.budget().reset_unlimited();
+
+    let bombadil = Address::generate(&e);
+    let frodo = Address::generate(&e);
+
+    let settings = default_governor_settings();
+    let (governor_address, token_address, votes_address) =
+        create_governor(&e, &bombadil, &settings);
+    let token_client = MockTokenClient::new(&e, &token_address);
+    let votes_client = TokenVotesClient::new(&e, &votes_address);
+    let governor_client = GovernorContractClient::new(&e, &governor_address);
+    let (outter_subcall_address, _) =
+        create_mock_subcall_contract_wasm(&e, &token_address, &governor_address);
+    let (inner_subcall_address, _) =
+        create_mock_subcall_contract_wasm(&e, &token_address, &governor_address);
+
+    // set intial votes
+    let frodo_votes: i128 = 10_000 * 10i128.pow(7);
+    token_client.mint(&frodo, &frodo_votes);
+    votes_client.deposit_for(&frodo, &frodo_votes);
+
+    // create a proposal
+    let (_, _, title, description) = default_proposal_data(&e);
+    let call_amount: i128 = 100 * 10i128.pow(7);
+    token_client.mint(&governor_address, &call_amount);
+    let calldata = Calldata {
+        contract_id: outter_subcall_address.clone(),
+        function: Symbol::new(&e, "sub_subcall"),
+        args: (inner_subcall_address.clone(), call_amount.clone()).into_val(&e),
+    };
+    let sub_calldata: Vec<SubCalldata> = vec![
+        &e,
+        SubCalldata {
+            contract_id: inner_subcall_address.clone(),
+            function: Symbol::new(&e, "subcall"),
+            args: (call_amount.clone(),).into_val(&e),
+            sub_auth: vec![
+                &e,
+                SubCalldata {
+                    contract_id: token_address,
+                    function: Symbol::new(&e, "transfer"),
+                    args: (
+                        governor_address.clone(),
+                        inner_subcall_address.clone(),
+                        call_amount.clone(),
+                    )
+                        .into_val(&e),
+                    sub_auth: vec![&e],
+                },
+            ],
+        },
+    ];
```

Comment:
do we know what happens if `sub_subcall` does not have a `require_auth` call?

Is the first auth needed?

We should maybe add a second test that verifies that case would work too? It would likely be an important documentation point.

Consider a Governor calling a contract that interacts with a AMM. Its not impossible for the middle contract to omit `require_auth` as they know the underlying contract will be requiring auth for the transfer already.